src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,297 @@
+                        parametersWithoutArgument[0], ImmutableArray<SyntaxNode>.Empty));
+                }
+
+                var parametersWithDefaultValue = parametersWithoutArgument
+                    .Where(parameterSymbol => parameterSymbol.HasExplicitDefaultValue)
+                    .ToImmutableArray();
+
+                return new MethodParametersInfo(
+                    parametersWithIdentifier.SelectAsArray(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, parameterAndArgument.arguments[0])),
+                    parametersWithVariableDeclarationArgument.SelectAsArray(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, parameterAndArgument.arguments[0])),
+                    parametersNeedGenerateDeclarations,
+                    parametersWithDefaultValue,
+                    parametersWithLiteralArgument.SelectAsArray(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, parameterAndArgument.arguments[0])));
+            }
+
+            private static bool ParameterNeedsGenerateDeclarationFilter(
+                AbstractInlineMethodRefactoringProvider<TInvocationSyntaxNode, TExpressionSyntax, TArgumentSyntax> inlineMethodRefactoringProvider,
+                ISyntaxFacts syntaxFacts,
+                SemanticModel semanticModel,
+                (IParameterSymbol parameterSymbol, ImmutableArray<SyntaxNode> arguments) parameterAndArguments,
+                CancellationToken cancellationToken)
+            {
+                var (parameterSymbol, arguments) = parameterAndArguments;
+                if (!parameterSymbol.IsParams && arguments.Length == 1)
+                {
+                    var argument = arguments[0];
+                    return IsExpressionSyntax(argument);
+                }
+
+                // Params array is special, if there are multiple arguments, then they needs to be put into a separate array.
+                if (parameterSymbol.IsParams)
+                {
+                    if (arguments.Length != 1)
+                    {
+                        // If there is no argument or multiple arguments, a array initializer expression should be
+                        // put into caller
+                        return true;
+                    }
+                    else
+                    {
+                        // If there is only one argument, it has 3 cases
+                        // 1. This argument is an array (identifier)
+                        // 2. This argument is an array (array creation expression)
+                        // 3. This argument is the single element in the array. (identifier or literal)
+                        // Generate the declaration for case 2 & 3
+                        var argument = arguments[0];
+                        var typeOfElement = (IArrayTypeSymbol)parameterSymbol.Type;
```

Comment:
how do you know it's an array?

Reply:
Because it is a param symbol so its type should be array. 
But this part of code is also not need because I tried to use the IOperation and hence this can be solved.